
# Resources and tools for epigenetics in Bioconductor

This section offers a summary of the main data structures that are implemented in [Bioconductor](https://bioconductor.org/) for dealing epigenomic data data.  `Omic` data is typically composed of three datasets: one containing the actual high-dimensional data of omic variables per individuals, annotation data that specifies the characteristics of the variables and phenotypic information that encodes the subject's traits of interest, covariates and sampling characteristics. Bioconductor can effectively deal with this data using (among others) `ExpressionSet` objects which is a data structure that contains the beta values of individuals at each CpG, their genomic information and the phenotypes of the individuals. Specific data is accessed, processed and analyzed with specific functions from diverse packages, conceived as methods acting on the `ExpressionSet` object. 

We will need these libraries:
```{r load_intro, warning=FALSE, message=FALSE}
library(BiocManager)
library(GenomicRanges)
library(GEOquery)
library(brgedata)
```


## Bioconductor
[Bioconductor](https://bioconductor.org/): Analysis and comprehension of high-throughput
genomic data

- Statistical analysis: large data, technological artifacts, designed
  experiments; rigorous
- Comprehension: biological context, visualization, reproducibility
- High-throughput
    - Sequencing: RNASeq, ChIPSeq, variants, copy number, ...
    - Microarrays: expression, SNP, ...
    - Flow cytometry, proteomics, images, ...

Packages, vignettes, work flows

- 2230 packages (Apr'23); including ...
- 'Software' packages -- statistical methods, pipelines, ...
- 'Annotation' packages -- static data bases of identifier maps,
gene models, pathways, etc; e.g., [TxDb.Hsapiens.UCSC.hg19.knownGene](https://bioconductor.org/packages/release/data/annotation/html/TxDb.Hsapiens.UCSC.hg19.knownGene.html)
- 'Experiment packages -- data sets used to illustrate software
      functionality, e.g., [airway](https://bioconductor.org/packages/release/data/experiment/html/airway.html)
- Discover and navigate via [biocViews](https://bioconductor.org/packages/release/bioc/html/biocViews.html)
- Package 'landing page'
    - Title, author / maintainer, short description, citation,
      installation instructions, ..., download statistics
- All user-visible functions have help pages, most with runnable
  examples
- 'Vignettes' an important feature in Bioconductor -- narrative
  documents illustrating how to use the package, with integrated code
- 'Release' (every six months) and 'devel' branches
- [Support site](https://support.bioconductor.org);
  [videos](https://www.youtube.com/user/bioconductor);
  [recent courses](https://bioconductor.org/help/course-materials/).
- [Common Workflows](https://bioconductor.org/packages/release/BiocViews.html#___Workflow)  

Package installation and use

- A package needs to be installed once, using the instructions on the
  package landing page (e.g., [DESeq2](https://bioconductor.org/packages/release/bioc/html/DESeq2.html)).

    ```{r install_2, eval=FALSE}
    require(BiocManager)
    install("DESeq2")
    
    # or
    
    BiocManager::install("DESeq2")
    ```

- older versions can be installed by

    ```{r install_old, eval=FALSE}
    BiocManager::install("DESeq2", version = "3.8")
    ```

- `install()` install _Bioconductor_ and [CRAN](https://cran.r-project.org/)  

- Github packages can be install by

    ```{r install_github, eval=FALSE}
    devtools::install_github("isglobal-brge/SNPassoc")
    ```

- Once installed, the package can be loaded into an R session

    ```{r require}
    library(GenomicRanges)
    ```

    and the help system queried interactively, as outlined above:

    ```{r help-bioc, eval=FALSE}
    help(package="GenomicRanges")
    vignette(package="GenomicRanges")
    vignette(package="GenomicRanges", "GenomicRangesHOWTOs")
    ?GRanges
    ```


## ExpressionSet and GenomicRatioSet (and others: GRanges)

`ExpressionSet`was one of the first implementations of Bioconductor to manage *omic* experiments. This figure illustrates how it is implemented

```{r eSetfig, fig.cap='ExpressionSet scheme.', out.width='100%', fig.align='center', echo=FALSE}
knitr::include_graphics('images/eSet_vs_dataframe.png')
```

Although its use is discouraged in Bioconductor's guidelines for the development of current and future packages (Summarized Experiments are preferred), most publicly data are available in this structure while future packages are still required to be able to upload and operate with it. This is the case of [GEO](https://www.ncbi.nlm.nih.gov/geo/) (Gene Expression Omnibus) which is an international public repository that hosts and freely distribute an extensive collection of high-throughput gene expression data generated through technologies like microarrays and RNA sequencing. 


### Gene Expression Omnibus (GEO) repository

Researchers from diverse fields can upload their data to GEO, making it accessible to the scientific community for further analysis and interpretation. One of the notable benefits of GEO is its role in validating results from other studies. By comparing and contrasting their findings with existing datasets, researchers can corroborate their conclusions and ensure the reproducibility of their experiments. This validation process is essential for building a robust foundation of knowledge in molecular biology and fostering scientific progress through collaborative efforts and data-driven insights.

GEO was initially designed to deal with transcriptomic data, but it can also be used as a repository for *epigenomic data*. For example, we can further investigate the methylation differences in cases diagnoses with Alzheimer's disease and controls using publicly available data. The study with accession number GSE80970, is a case/control ($N=148/138$) study of epigenomic data, measured with the Illumina Infinium Human 450K Methylation Array. Data is retrieved directly from the GEO website with the accession number. 

Data for specific accessions in GEO can be downloaded in `R` using the `getGEO` of the Bioconductor's package `GEOquery`

```{r geoquery}
library(GEOquery)

if(!dir.exists("c:/tmp"))
  dir.create("c:/tmp")

gsm <- getGEO("GSE80970", destdir = "c:/tmp")
gsm <- gsm[[1]]
```


This function downloads a data file called `GSE80970\_series\_matrix.txt.gz` that is retrieved automatically in an `ExpressionSet` object (this objects will be explain latter) that we have named `gms.expr`

```{r geoquery2}
class(gsm)
gsm
```

An object of class `ExpressionSet` stores different tables including the expression profiles for each probe and subject `assayData`, phenotype data with traits measurements and covariates of interest `pData`, and feature data with information about the probe's used in the expression (or methylation) array `fData` (e.g. annotation). Specific data is retrieved using the necessary functions. In particular,  `exprs ()` and  `phenoData ()` extract data tables for subjects' expression levels and phenotypes/covariates, respectively.  There are three other slots `protocolData`, `experimentData` and `annotation` (that uses Bioconductor databases as annotation data - i.e. it does not requires `fData`) that specify equipment-generated information about protocols, resulting publications and the platform on which the samples were assayed. Methods are implemented to extract the data from each slot of the object. 

Let us see how it works. The function `exprs ()` extracts the epigenetic data in a matrix where subjects are columns and probes are rows



```{r getdata}
#get epigenetic data 
expr <- exprs(gsm)
dim(expr)
expr[1:10, 1:5]
```

Phenotypes and/or covariates can be accessed using `phenoData ()` function

```{r getPheno}
#get phenotype data
pheno <- pData(phenoData(gsm))
head(pheno)

#main variable used to be stored in: 
table(pheno$characteristics_ch1)
```

Note that `phenoData()` retrieves the subjects' phenotypes in an  `AnnotatedDataFrame` object  which is converted to a data.frame by the function `pData()`    



We can the, for instance, look at the CpGs located in the APP gene. To this end, the function `fData ()` is used

```{r app}

#get the annotation
annot <- fData(gsm)

colnames(annot)

#access to the names of genes in UCSC format
geneIDs <- annot$UCSC_RefGene_Name
genesIDs <- as.character(unlist(geneIDs))

selAPP <- grep("APP;APP",genesIDs)
length(selAPP)
```

After that we can run different analyses. Also, several packages in Bioconductor have as input `ExpressionSet` objects. 


### `SummarizedExperiment`

The `SummarizedExperiment` class is a comprehensive data structure that can be used to store expression and methylation data from microarrays or read counts from RNA-seq experiments, among others. A `SummarizedExperiment` object contains slots for one or more \emph{omic} datasets, feature annotation\index{subjectindex}{Annotation} (e.g. genes, transcripts, SNPs, CpGs), individual phenotypes and experimental details, such as laboratory and experimental protocols. In a `SummarizedExperiment`, the rows of *omic* data are features and columns are subjects. 

![SummarizedExperiment](images/sumExp.svg)


Information is coordinated across the object's slots. For instance, subsetting samples in the assay matrix automatically subsets them in the phenotype metadata. A `SummarizedExperiment`object is easily manipulated and constitutes the input and output of many of Bioconductor's methods. 

Data is retrieved from a `SummarizedExperiment` by using specific methods or accessors. We illustrate the functions with  `brge_methy` which includes real methylation data and is available from the Bioconductor's  `brgedata` package. The data is made available by loading of the package

```{r summexp_example, warning=FALSE, message=FALSE}
library(brgedata)
brge_methy
extends("GenomicRatioSet")
```

The function `extends()` shows that the data has been encoded in  an object of `GenomicRatioSet` class, which is an extension of the more primitive classes `RangedSummarizedExperiment` and `SummarizedExperiment`. `brge_methy` illustrates a typical object within Bioconductor's framework, as it is a structure that inherits different types of classes in an established hierarchy. For each class, there are specific methods which are properly inherited across classes. For instance, in our example, `SummarizedExperiment` is `brge_methy`'s most primitive *omic* class and, therefore, all the methods of `SummarizedExperiment` apply to it. In particular, the methylation data that is stored in the object can be extracted with the function `assay()`

```{r get_assays}
betas <- assay(brge_methy)
betas[1:5, 1:4]
```

The assay slot of a `SummarizedExperiment` object can contain any type of data (i.e. numeric, character, factor...), structure or large on-disk representations, such as a HDF5Array. Feature annotation data is accessed with the function `rowData()`:
  
```{r get_features}
rowData(brge_methy)[,2:5]
```

which returns a data.frame object. In our example, it contains the sequences and the genes associated with the CpG probes, among other information. 


### `Genomic Ranges (GRanges)`

The Bioconductor's package `GenomicRanges` aims to represent and manipulate the genomic annotation of molecular *omic* data under a reference genome. It contains functions to select specific regions and perform operations with them [@lawrence2013software]. Objects of `GRanges` class are important to annotate and manipulate genomic, transcriptomic and methylomic data. In particular, they are used in conjunction with `SummarizedExperiment`, within the `RangedSummarizedExperiment` class that is explained in the following section. 

Annotation data refers to the characteristics of the variables in the high-dimensional data set. In particular for *omic* data relative to DNA structure and function, each variable may be given a location in a reference genome. While not two genomes are identical, the construction of a reference genome allows the mapping of specific characteristics of individual genomes to a common ground where they can be compared. The reference genome defines a coordinate system: “chromosome id” and “position along the chromosome”. For instance, a position such as chr10:4567-5671 would represent the 4567th to the 5671st base pair on the reference's chromosome 10.

The main functionalities implemented `GenomicRanges` are methods on `GRanges` objects. Objects are created by the function `GRanges`, minimum requirements are the genomic positions given by the chromosome `seqnames` and base pair coordinates  `ranges`. Other metadata (e.g. variables) can be added to provide further information about each segment. 

We illustrate `GenomicRanges` creating 8 segments on either chr1 or chr2, each with defined start and end points. We add strand information, passed through the argument `strand`, to indicate the direction of each sequence. We also add a hypothetical variable `disease` that indicates whether asthma or obesity have been associated with each interval

```{r createGR}
library(GenomicRanges)
gr <- GRanges(seqnames=c(rep("chr1", 4), rep("chr2", 4)),
              ranges = IRanges(start = c(1000, 1800, 5300, 7900,
                                         1300, 2100, 3400, 6700),
                               end =c(2200, 3900, 5400, 8100,
                                      2600, 3300, 4460, 6850)),
              strand = rep(c("+", "-"), 4),
              disease = c(rep("Asthma",4), rep("Obesity",4)))
gr
```

`gr` is our constructed object of class `GRanges`. The \Rclass{GRanges} object responds to the usual array and subset extraction given by squared parentheses

```{r gr1}
gr[1]
```

However, there are also specific functions to access and modify information. For instance, `seqnames()` extract the chromosomes defined in our examples, whose first element can be redefined accordingly:
  
```{r gr2}
seqnames(gr)
seqnames(gr)[1] <- "chr2"
gr
``` 

This is important to annotation using different system. NCBI encodes chromosomes as 1, 2, 3, ...; while UCSC uses chr1, chr2, ... The chromosome style can be changed using

```{r gr2a}
seqlevelsStyle(gr) <- "NCBI"
gr
``` 

Let's us write back the UCSC style

```{r gr2b}
seqlevelsStyle(gr) <- "UCSC"
gr
``` 


Additional information can be added to the current object as a new field of a list

```{r gr3}
gr$gene_id <- paste0("Gene", 1:8)
gr
```

`GenomicRanges` provides different methods to perform arithmetic with the ranges, see `?GRanges` for a full list. For instance, with `shift()` an interval is moved a given base-pair distance and with `flank()` the interval is stretched  

```{r gr4}
#shift: move all intervals 10 base pair towards the end
GenomicRanges::shift(gr, 10)

#shift: move each intervals individually
GenomicRanges::shift(gr, seq(10,100, length=8))

#flank:  recover regions next to the input set. 
#        For a 50 base stretch upstream (negative value for
#        downstream)
flank(gr, 50)
```

`GenomicRanges` also includes methods for aggregating and summarizing `GRanges`objects. The functions `reduce()`, `disjoint()` and `coverage()` are most useful. `disjoin()`, for instance, reduces the intervals into the smallest set of unique, non-overlapping pieces that make up the original object. It is strand-specific by default, but this can be avoided with `ignore.strand=TRUE`.

```{r gr5}
disjoin(gr, ignore.strand=TRUE)
```

`reduce()` creates the smallest range set of unique, non-overlapping intervals. Strand information is also taken into account by default and can also be turned off

```{r gr6}
GenomicRanges::reduce(gr, ignore.strand=TRUE)
```

`coverage()` summarizes the times each base is covered by an interval

```{r gr7}
coverage(gr)
```

It is also possible to perform operations between two different `GRanges` objects. For instance, one may be interested in knowing the intervals that overlap with a targeted region:

```{r gr8}
target <- GRanges(seqnames="chr1", 
                  range=IRanges(start=1200, 4000))
target
gr.ov <- findOverlaps(target, gr)
gr.ov
```

To recover the overlapping intervals between `gr` and `target` we can run

```{r gr9}
gr[subjectHits(gr.ov)]
``` 

or

```{r gr10}
subsetByOverlaps(gr, target)
```

Other operations can be found [here](https://bioconductor.org/help/course-materials/2014/SeattleFeb2014/). 

### `RangedSummarizedExperiment`

`SummarizedExperiment` is extended to `RangedSummarizedExperiment`, a child class that contains the annotation data of the features in a `GenomicRanges` object. In our epigenomic example, the second most primitive class of `brge_methy` object with *omic* functionality, after `SummarizedExperiment`, is `RangedSummarizedExperiment`. Annotation data, with variable names given by

```{r names_rowranges}
names(rowData(brge_methy))
```

can be obtained in a `GRanges` object, for a given variable. For instance, metadata of CpG genomic annotation and neighboring genes is obtained using array syntax

```{r rowranges}
rowRanges(brge_methy)[, "genes"]
```

Subject data can be accessed entirely in a single data.frame or a variable at the time. The entire subject (phenotype) information is retrieved with the function `colData()`:

```{r sample_metadata}
colData(brge_methy)
```

The list symbol `$` can be used, for instance, to obtain the sex of the individuals

```{r sample_metadata_variable}
brge_methy$sex
```

Subsetting the entire structure is also possible following the usual array syntax. For example, we can select only males from the `brge_methy` dataset

```{r get_males}
brge_methy[, brge_methy$sex == "male"]
```

The `metadata()` function retrieves experimental data

```{r get_metadata}
metadata(brge_methy)
```

which in our case is empty.

