# Epigenome-wide association study (EWAS)
In this chapter we will run an Epigenome-Wide Association Study (EWAS)
A file with the slides can be downloaded [here](./Slides/ISG_summer_school_EWAS3.pptx)

At the end of the practice, please answer these questions:

1. Which is the lambda of the unadjusted EWAS of current smoking? How does it change in adding covariates and surrogate variables?
2. How many CpGs are associated with current smoking (after False Discovery Rate â€“ FDR - correction) in the model adjusted by covariates and sva?
3. How many of the FDR CpGs show higher methylation and how many lower methylation?
4. Which is the top 1 CpG? In which chromosome is located?


## Install and load packages
Today we will load these packages
```{r libload}
library(Biobase)# to be able to access and modify data in the ExpressionSet
library(meffil) # to run the EWAS
library(ggplot2)# plots
library(ggrepel)
library(GenomicRanges) # prepare db for Manhattan plot
library(karyoploteR) #  Manhattan plot 
```

## Load dataset and extract the phenotypes and methylation matrix

We begin loading the output from the last step, which is the ExpressionSet that
contains the normalised betas together with the phenotypes.
```{r loaddataset}
eset.cohort1<-readRDS("./data/DAY3/Input/GSE42861_norm_cohort1.Rds") 
pheno.cohort1<-pData(eset.cohort1)
dim(pheno.cohort1)
head(pheno.cohort1)

methyl.cohort1<-exprs(eset.cohort1)
dim(methyl.cohort1) 
methyl.cohort1[1:5,1:5]

table(eset.cohort1$Smoking)
```

## Run EWAS
### Data preparation
Here we will show an example to run the analysis testing never smokers against 
current smokers. Nevertheless, the same code can be applied to run other
analyses testing never vs former or current vs former. 

The first step is to subset the ExpressionSet with only the samples that we need.
In this case, we only keep the never smokers and the current smokers,
getting rid of the former smokers. 
```{r subset}
current <- pheno.cohort1[pheno.cohort1$Smoking %in% c('never','current'),]
dim(current)
eset.current <- eset.cohort1[,rownames(current)] #subset the eset
eset.current
table(eset.current$Smoking)
```

Extract methylation
```{r methylcurrent}
methyl.current<-exprs(eset.current)
methyl.current[1:5,1:5]
```

Extract phenodata with never and current smokers
```{r phenocurrent}
pheno.current<-pData(eset.current)
pheno.current$Smoking<-as.character(pheno.current$Smoking)
pheno.current$Smoking<-as.factor(pheno.current$Smoking)
```

Check the levels. We want "never" to be the reference level
```{r levels}
levels(pheno.current$Smoking)
pheno.current$Smoking <- relevel(pheno.current$Smoking, "never")
levels(pheno.current$Smoking)
```
Select the exposure variable (Smoking)
```{r variable}
variable <- pheno.current$Smoking
class(variable)
```
Select the covariates of interest for the EWAS. We are interested in sex, age 
and cell type proportions. 
Check the class of these covariates before running the analysis.
```{r covs}
class(pheno.current$Age)
class(pheno.current$Sex)
class(pheno.current$Bcell)
class(pheno.current$CD4T)
class(pheno.current$CD8T)
class(pheno.current$Mono)
class(pheno.current$Neu)
class(pheno.current$NK)
pheno.current$Sex<-as.factor(pheno.current$Sex)
covariates <- pheno.current[,c("Age","Sex","Bcell","CD4T","CD8T","Mono","Neu","NK")]
```

Check order of the samples between the pheno and the methylation matrix. 
If you do not order samples as they are in the ExpressionSet, you could 
incorrectly assign the values of the variables to the samples and therefore also
to the methylation.
```{r check}
table(ifelse(rownames(pheno.current)==colnames(methyl.current),"Matched","--NOT MATCHED--"))
```

### EWAS
To run the EWAS we will use the function meffil.ewas from meffil R package. We need as arguments:

- beta: Methylattion matrix
- variable: Vector with the exposure variable. 
- covariates: A dataframe with the covariates to include in the regression model
- rlm: As we want to run a robust linear regression model, we need to specify 
"TRUE" in the argument "rlm"
- winsorize.pct: To reduce the impact of severe outliers in the DNA methylation 
data, we will winsorize the methylation beta values (1%), with 0.5% at upper and
lower ends of the distribution [@ghoshvogt2012]
- sva (by default TRUE). This function will apply Surrogate Variable Analysis 
(SVA) to the methylation levels and covariates and include the resulting variables
as covariates in the regression model

```{r ewas, warning=FALSE, message=FALSE}
ewas.current <- meffil.ewas(methyl.current,
                            variable=variable,
                            covariates=covariates,
                            rlm=TRUE,
                            winsorize.pct=0.05)
```

### Results
We obtain an object with several information
```{r summary1, warning=FALSE, message=FALSE}
summary(ewas.current)
```
We are interested in the EWAS results. This function calculated the results  
1) without adjusting for covariates, 2) adjusting for the covariates we specified above
and also 3) adjusting for the covariates and the surrogate variables to correct the batch (noise)
```{r summary2}
summary(ewas.current$analyses)
```
We will use the results from the analysis adjusted for SVA
```{r res}
res.current<-ewas.current$analyses$sva
res.current<-res.current$table
```
Create a column with the Probe ID, which is in the rows of the results dataframe
```{r probeid}
res.current$probeID<-rownames(res.current)
```

We order the results by p.value
```{r order_res}
res.current.ord<-res.current[order(res.current$p.value),]
head(res.current.ord)
colnames(res.current.ord)
```
The column names stand for:

- p.value: pval significance of the association
- fdr: p.value corrected by the FALSE DISCOVERY RATE method
- p.holm: p.value corrected by Holm-Bonferroni method (we are not going to use it)
- t.statistic 
- coefficient: coefficient of the association. In this case is odd ratio (OR), which is a statistical measure to quantify the strength of association of a binary variable (the effect or beta)
- coefficient.ci.high: confidence interval up 
- coefficient.ci.low: confindence interval down
- coefficient.se:  standard error of the coefficient
- n: sample size included in the analysis
- chromosome: chromosome in which the CpG is located in the genome (hg19)   
- position: position of the CpG in the genome (hg19)         

**Significant hits**

Our results were corrected for multiple testing using FDR method. Significance 
was defined at FDR p-value < 0.05. We tested 37,842 CpGs. 
```{r totalCpGs}
dim(res.current.ord)
```

Significant CpGs with a FDR p-value <0.05
```{r significantFDR}
res.current.sig<-res.current.ord[res.current.ord$fdr < 0.05,]
dim(res.current.sig)
head(res.current.sig)
```
Number of significant CpGs with a positive or negative effect
```{r DirectionEffect}
dim(res.current.sig[res.current.sig$coefficient <0,]) 
dim(res.current.sig[res.current.sig$coefficient >0,]) 

```

We save the results
```{r saveewas, eval=FALSE}
write.table(res.current.ord,
            "./data/DAY3/Output/Results.EWAS.cohort1.NeverVScurrent.txt",
            sep=",")

```

### Report

To explore the results, we will create a report using meffil.ewas.parameters() and meffil.ewas.summary() functions from meffil R package. 

This report contains: 

- Sample characteristics
- Covariate associations: Associations between the exposure variable (Smoking) and the covariates. 
- Lambdas and QQplots to examine inflation (none, all cov, sva)
- Manhattan plots
- Significant CpG sites. As threshold we will indicate FDR=0.05.
- Specific CpGs Box plot: To observe the methylation differences between Never VS Current smokers, you can specify the CpGs that you are interested in. Today we will select our top significant CpG "cg05575921" that is annotated to AHRR gene and it is well known for its association with tabbacco. 

```{r report, eval=FALSE}
ewas.parameters <- meffil.ewas.parameters(max.plots = 1,model="sva", sig.threshold = 3.86e-04) #FDR (0.05) = p.value (3.86e-04)
candidate.site <- c("cg05575921")
ewas.summary <- meffil.ewas.summary(ewas.current,
                                    methyl.current,
                                    selected.cpg.sites=candidate.site,
                                    parameters=ewas.parameters)								
meffil.ewas.report(ewas.summary,
                   output.file="./data/DAY3/Output/Report.EWAS.cohort1.NeverVScurrent.html")
```
The report can be found [here](./data/DAY3/Output/Report.EWAS.cohort1.NeverVScurrent.html)

### Plots

**Volcano plot**

The plot that you should obtain can be found [here](./data/DAY3/Output/VolcanoPlot.EWAS.cohort1.NeverVScurrent.png)
```{r volcano, warnings=FALSE, message=FALSE}
res.current.ord$diffexpressed <- "NO"
res.current.ord$diffexpressed[res.current.ord$coefficient > 0
                              & res.current.ord$fdr <0.05] <- "POSITIVE"
res.current.ord$diffexpressed[res.current.ord$coefficient < 0 
                              & res.current.ord$fdr <0.05] <- "NEGATIVE"

p <- ggplot(data=res.current.ord, aes(x=res.current.ord$coefficient, y=-log10(res.current.ord$p.value), col=res.current.ord$diffexpressed)) +
  xlim(c(-0.3,0.3))+ ylim(c(0,35)) + 
  geom_point(size = 1.5) + theme_minimal() +
  labs(title = " ", x = "beta", y = "-log10(P-value)", colour = "Effect") + 
  theme(axis.title = element_text(size = 14, color = "black",vjust=0.5)) + 
  theme(plot.title = element_text(size = 14,face="bold",color="black",
                                  hjust= 0.5, vjust=0.5)) + 
  theme(legend.title = element_text(color = "black", size = 14))


# Add lines as before...
p2 <- p + geom_hline(yintercept=c(-log10(3.86e-04)), col=c("red"),
                     linetype = "dashed") + 
  theme(axis.text = element_text(size = 14)) #thres FDR (0.05) = p.val 3.86-04
mycolors<-c("#157F8D","#AF8D9B", "grey")
names(mycolors) <- c("POSITIVE", "NEGATIVE", "NO")

p3 <- p2 + scale_colour_manual(values = mycolors)

p3

```


**Manhattan plot**

The plot that you should obtain can be found [here](./data/DAY3/Output/ManhattanPlot.EWAS.cohort1.NeverVScurrent.png)
```{r manhattan, out.width="100%"}
# Create a dataframe with chr, start, end and pval
df.current<-res.current.ord[,c("chromosome","position","p.value")]
head(df.current)
df.current$start<-df.current$position 
df.current$end<-df.current$position + 1
colnames(df.current)<-c("chr","position","p.value","start","end")
df.current<-df.current[,c("chr","start","end","p.value")]

# Create GRanges object needed 
df.GRanges<-makeGRangesFromDataFrame(df.current, keep.extra.columns = TRUE)

#top.cpgs <- subset(df.GRanges, p.value<1e-30)

kp <- plotKaryotype(plot.type=4, labels.plotter = NULL)
kp <- kpAddChromosomeNames(kp, cex=0.6, srt=45)
kp <- kpPlotManhattan(kp, data=df.GRanges,pval=df.current$p.value, ymax=40,genomewideline =3.42)
kp <- kpAxis(kp, ymin=0, ymax=40, cex=0.7)
#kpText(kp, data = top.cpgs, labels = names(top.cpgs), ymax=10, pos=3)

```

