<!DOCTYPE html>
<html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<title>Chapter 5 Biological interpretation of findings | Introduction to Epigenome-Wide Association Studies (EWAS)</title>
<meta name="author" content="Juan R. González, Mariona Bustamante, Sofía Aguilar &amp; Laura Balagué">
<meta name="description" content="In this chapter we will explain the steps for the biological interpretation of the findings. A file with the slides can be downloaded here At the end of the practice, please answer these...">
<meta name="generator" content="bookdown 0.34 with bs4_book()">
<meta property="og:title" content="Chapter 5 Biological interpretation of findings | Introduction to Epigenome-Wide Association Studies (EWAS)">
<meta property="og:type" content="book">
<meta property="og:url" content="https://isglobal-brge.github.io/course_methylation/biological-interpretation-of-findings.html">
<meta property="og:description" content="In this chapter we will explain the steps for the biological interpretation of the findings. A file with the slides can be downloaded here At the end of the practice, please answer these...">
<meta name="twitter:card" content="summary">
<meta name="twitter:title" content="Chapter 5 Biological interpretation of findings | Introduction to Epigenome-Wide Association Studies (EWAS)">
<meta name="twitter:description" content="In this chapter we will explain the steps for the biological interpretation of the findings. A file with the slides can be downloaded here At the end of the practice, please answer these...">
<!-- JS --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/fuse.js/6.4.6/fuse.js" integrity="sha512-zv6Ywkjyktsohkbp9bb45V6tEMoWhzFzXis+LrMehmJZZSys19Yxf1dopHx7WzIKxr5tK2dVcYmaCk2uqdjF4A==" crossorigin="anonymous"></script><script src="https://kit.fontawesome.com/6ecbd6c532.js" crossorigin="anonymous"></script><script src="libs/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<link href="libs/bootstrap-4.6.0/bootstrap.min.css" rel="stylesheet">
<script src="libs/bootstrap-4.6.0/bootstrap.bundle.min.js"></script><script src="libs/bs3compat-0.5.0/transition.js"></script><script src="libs/bs3compat-0.5.0/tabs.js"></script><script src="libs/bs3compat-0.5.0/bs3compat.js"></script><link href="libs/bs4_book-1.0.0/bs4_book.css" rel="stylesheet">
<script src="libs/bs4_book-1.0.0/bs4_book.js"></script><link href="libs/pagedtable-1.1/css/pagedtable.css" rel="stylesheet">
<script src="libs/pagedtable-1.1/js/pagedtable.js"></script><link rel="icon" type="image/png" href="images/Logo_Sportomics_AI.jpg">
<script src="https://cdnjs.cloudflare.com/ajax/libs/autocomplete.js/0.38.0/autocomplete.jquery.min.js" integrity="sha512-GU9ayf+66Xx2TmpxqJpliWbT5PiGYxpaG8rfnBEk1LL8l1KGkRShhngwdXK1UgqhAzWpZHSiYPc09/NwDQIGyg==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mark.js/8.11.1/mark.min.js" integrity="sha512-5CYOlHXGh6QpOFA/TeTylKLWfB3ftPsde7AnmhuitiTX4K5SqCLBeKro6sPS8ilsz1Q4NRx3v8Ko2IBiszzdww==" crossorigin="anonymous"></script><!-- CSS --><style type="text/css">
    
    div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  </style>
<style type="text/css">
    /* Used with Pandoc 2.11+ new --citeproc when CSL is used */
    div.csl-bib-body { }
    div.csl-entry {
      clear: both;
        }
    .hanging div.csl-entry {
      margin-left:2em;
      text-indent:-2em;
    }
    div.csl-left-margin {
      min-width:2em;
      float:left;
    }
    div.csl-right-inline {
      margin-left:2em;
      padding-left:1em;
    }
    div.csl-indent {
      margin-left: 2em;
    }
  </style>
<link rel="stylesheet" href="style.css">
</head>
<body data-spy="scroll" data-target="#toc">

<div class="container-fluid">
<div class="row">
  <header class="col-sm-12 col-lg-3 sidebar sidebar-book"><a class="sr-only sr-only-focusable" href="#content">Skip to main content</a>
    
    <img id="cover" class="d-none d-lg-block" src="images/isglobal-logo.png" width="200"><div class="d-flex align-items-start justify-content-between">
      <h1>
        <a href="index.html" title="">Introduction to Epigenome-Wide Association Studies (EWAS)</a>
      </h1>
      <button class="btn btn-outline-primary d-lg-none ml-2 mt-1" type="button" data-toggle="collapse" data-target="#main-nav" aria-expanded="true" aria-controls="main-nav"><i class="fas fa-bars"></i><span class="sr-only">Show table of contents</span></button>
    </div>

    <div id="main-nav" class="collapse-lg">
      <form role="search">
        <input id="search" class="form-control" type="search" placeholder="Search" aria-label="Search">
</form>

      <nav aria-label="Table of contents"><h2>Table of contents</h2>
        <ul class="book-toc list-unstyled">
<li><a class="" href="index.html">Introduction to Epigenome-Wide Association Studies (EWAS)</a></li>
<li><a class="" href="resources-and-tools-for-epigenetics-in-bioconductor.html"><span class="header-section-number">1</span> Resources and tools for epigenetics in Bioconductor</a></li>
<li><a class="" href="pre-processing-of-dna-methylation-data.html"><span class="header-section-number">2</span> Pre-processing of DNA methylation data</a></li>
<li><a class="" href="epigenome-wide-association-study-ewas.html"><span class="header-section-number">3</span> Epigenome-wide association study (EWAS)</a></li>
<li><a class="" href="meta-analysis-of-epigenome-wide-association-studies-metaewas.html"><span class="header-section-number">4</span> Meta-analysis of epigenome-wide association studies (metaEWAS)</a></li>
<li><a class="active" href="biological-interpretation-of-findings.html"><span class="header-section-number">5</span> Biological interpretation of findings</a></li>
<li><a class="" href="references.html">References</a></li>
</ul>

        <div class="book-extra">
          
        </div>
      </nav>
</div>
  </header><main class="col-sm-12 col-md-9 col-lg-7" id="content"><div id="biological-interpretation-of-findings" class="section level1" number="5">
<h1>
<span class="header-section-number">5</span> Biological interpretation of findings<a class="anchor" aria-label="anchor" href="#biological-interpretation-of-findings"><i class="fas fa-link"></i></a>
</h1>
<p>In this chapter we will explain the steps for the biological interpretation of
the findings. A file with the slides can be downloaded <a href="./Slides/ISG_summer_school_EWAS5.pdf">here</a></p>
<p>At the end of the practice, please answer these questions:</p>
<ol style="list-style-type: decimal">
<li>Which is the top 1 enriched GO term in current smokers? and in former smokers?</li>
<li>Which are the enriched KEGG pathways in current smokers? and in former smokers?</li>
<li>Which are the enriched tissues in current smokers? and informer smokers?</li>
</ol>
<p>We will need to load some libraries:</p>
<div class="sourceCode" id="cb217"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va">IlluminaHumanMethylation450kanno.ilmn12.hg19</span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="https://yulab-smu.top/biomedical-knowledge-mining-book/">clusterProfiler</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va">org.Hs.eg.db</span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="https://yulab-smu.top/biomedical-knowledge-mining-book/">enrichplot</a></span><span class="op">)</span></span></code></pre></div>
<div id="load-metaewas-results" class="section level2" number="5.1">
<h2>
<span class="header-section-number">5.1</span> Load MetaEWAS results<a class="anchor" aria-label="anchor" href="#load-metaewas-results"><i class="fas fa-link"></i></a>
</h2>
<p>We begin loading the Never vs Current FDR significant metaEWAS results</p>
<div class="sourceCode" id="cb218"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">metaEWAS.FDRsig</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/utils/read.table.html">read.table</a></span><span class="op">(</span>file <span class="op">=</span> <span class="st">"./data/DAY5/Input/MA.Smoking.NeverVScurrent.FDRsig.txt"</span>,</span>
<span>                              sep<span class="op">=</span><span class="st">" "</span>,</span>
<span>                              header <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/head.html">head</a></span><span class="op">(</span><span class="va">metaEWAS.FDRsig</span><span class="op">)</span></span></code></pre></div>
<div data-pagedtable="false">
<script data-pagedtable-source type="application/json">
{"columns":[{"label":[""],"name":["_rn_"],"type":[""],"align":["left"]},{"label":["probe"],"name":[1],"type":["chr"],"align":["left"]},{"label":["coef_cohort1"],"name":[2],"type":["dbl"],"align":["right"]},{"label":["se_cohort1"],"name":[3],"type":["dbl"],"align":["right"]},{"label":["P_VAL_cohort1"],"name":[4],"type":["dbl"],"align":["right"]},{"label":["N_for_probe_cohort1"],"name":[5],"type":["int"],"align":["right"]},{"label":["coef_cohort2"],"name":[6],"type":["dbl"],"align":["right"]},{"label":["se_cohort2"],"name":[7],"type":["dbl"],"align":["right"]},{"label":["P_VAL_cohort2"],"name":[8],"type":["dbl"],"align":["right"]},{"label":["N_for_probe_cohort2"],"name":[9],"type":["int"],"align":["right"]},{"label":["coef_cohort3"],"name":[10],"type":["dbl"],"align":["right"]},{"label":["se_cohort3"],"name":[11],"type":["dbl"],"align":["right"]},{"label":["P_VAL_cohort3"],"name":[12],"type":["dbl"],"align":["right"]},{"label":["N_for_probe_cohort3"],"name":[13],"type":["int"],"align":["right"]},{"label":["N"],"name":[14],"type":["int"],"align":["right"]},{"label":["coef.fe"],"name":[15],"type":["dbl"],"align":["right"]},{"label":["se.fe"],"name":[16],"type":["dbl"],"align":["right"]},{"label":["p.fe"],"name":[17],"type":["dbl"],"align":["right"]},{"label":["q.fe"],"name":[18],"type":["dbl"],"align":["right"]},{"label":["het.p.fe"],"name":[19],"type":["dbl"],"align":["right"]},{"label":["i2.fe"],"name":[20],"type":["dbl"],"align":["right"]},{"label":["h2.fe"],"name":[21],"type":["dbl"],"align":["right"]},{"label":["fdr"],"name":[22],"type":["dbl"],"align":["right"]}],"data":[{"1":"cg05575921","2":"-0.15723240","3":"0.010576833","4":"3.449766e-31","5":"188","6":"-0.14101120","7":"0.011691620","8":"2.420781e-23","9":"176","10":"-0.15591006","11":"0.010843238","12":"1.583696e-28","13":"166","14":"530","15":"-0.15198539","16":"0.006355165","17":"2.125374e-126","18":"1.2581465","19":"0.5330856","20":"0.00000","21":"0.6290732","22":"8.042838e-122","_rn_":"8751"},{"1":"cg21566642","2":"-0.11908682","3":"0.009286429","4":"8.297569e-26","5":"188","6":"-0.11343605","7":"0.009707079","8":"2.230469e-22","9":"176","10":"-0.10619119","11":"0.009754165","12":"5.891175e-20","13":"166","14":"530","15":"-0.11311143","16":"0.005528416","17":"4.893221e-93","18":"0.9184917","19":"0.6317599","20":"0.00000","21":"0.4592459","22":"9.258464e-89","_rn_":"29988"},{"1":"cg03636183","2":"-0.07262148","3":"0.005700836","4":"1.396044e-25","5":"188","6":"-0.06041403","7":"0.006198676","8":"2.073011e-17","9":"176","10":"-0.07531316","11":"0.006123817","12":"1.881683e-23","13":"166","14":"530","15":"-0.06967483","16":"0.003461447","17":"4.127409e-90","18":"3.3469160","19":"0.1875972","20":"40.24350","21":"1.6734580","22":"5.206314e-86","_rn_":"5907"},{"1":"cg01940273","2":"-0.07652157","3":"0.006414607","4":"1.992495e-23","5":"188","6":"-0.07737718","7":"0.006436027","8":"3.038132e-23","9":"176","10":"-0.06864217","11":"0.006436418","12":"2.091962e-19","13":"166","14":"530","15":"-0.07418574","16":"0.003711781","17":"7.209404e-89","18":"1.1202947","19":"0.5711249","20":"0.00000","21":"0.5601474","22":"6.820456e-85","_rn_":"3198"},{"1":"cg23576855","2":"-0.16467216","3":"0.014655727","4":"1.392884e-21","5":"188","6":"-0.14054494","7":"0.016847122","8":"6.601338e-14","9":"176","10":"-0.18243448","11":"0.013150485","12":"2.632126e-27","13":"166","14":"530","15":"-0.16594019","16":"0.008463186","17":"1.339678e-85","18":"3.8529221","19":"0.1456628","20":"48.09135","21":"1.9264611","22":"1.013922e-81","_rn_":"32409"},{"1":"cg05951221","2":"-0.09316890","3":"0.007577530","4":"2.110764e-24","5":"188","6":"-0.08917528","7":"0.008786733","8":"1.961694e-18","9":"176","10":"-0.08449224","11":"0.007639413","12":"2.190787e-20","13":"166","14":"530","15":"-0.08895022","16":"0.004588173","17":"9.965739e-84","18":"0.6511411","19":"0.7221152","20":"0.00000","21":"0.3255706","22":"6.285392e-80","_rn_":"9317"}],"options":{"columns":{"min":{},"max":[10]},"rows":{"min":[10],"max":[10]},"pages":{}}}
  </script>
</div>
</div>
<div id="annotation-of-cpg-sites" class="section level2" number="5.2">
<h2>
<span class="header-section-number">5.2</span> Annotation of CpG sites<a class="anchor" aria-label="anchor" href="#annotation-of-cpg-sites"><i class="fas fa-link"></i></a>
</h2>
<p>Once we have obtained a list with the CpGs that are significant in our analysis,
we need to locate them in the genome and try to know which structures
surround them. The annotation consists in obtaining this information:</p>
<p>First we load the annotation from <code>IlluminaHumanMethylation450kanno.ilmn12.hg19</code>
R package and select the columns of interest</p>
<div class="sourceCode" id="cb219"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/data.html">data</a></span><span class="op">(</span><span class="st">"IlluminaHumanMethylation450kanno.ilmn12.hg19"</span><span class="op">)</span></span>
<span><span class="va">annotation.table</span><span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/minfi/man/getAnnotation.html">getAnnotation</a></span><span class="op">(</span><span class="va">IlluminaHumanMethylation450kanno.ilmn12.hg19</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/dim.html">dim</a></span><span class="op">(</span><span class="va">annotation.table</span><span class="op">)</span></span></code></pre></div>
<pre><code>## [1] 485512     33</code></pre>
<div class="sourceCode" id="cb221"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">annotation.table</span><span class="op">&lt;-</span><span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/as.data.frame.html">as.data.frame</a></span><span class="op">(</span><span class="va">annotation.table</span><span class="op">[</span>,<span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"chr"</span>,<span class="st">"pos"</span>,<span class="st">"strand"</span>,</span>
<span>                                                    <span class="st">"Name"</span>,<span class="st">"Islands_Name"</span>,</span>
<span>                                                    <span class="st">"Relation_to_Island"</span>,</span>
<span>                                                    <span class="st">"UCSC_RefGene_Name"</span>,</span>
<span>                                                    <span class="st">"UCSC_RefGene_Group"</span><span class="op">)</span><span class="op">]</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/head.html">head</a></span><span class="op">(</span><span class="va">annotation.table</span><span class="op">)</span></span></code></pre></div>
<div data-pagedtable="false">
<script data-pagedtable-source type="application/json">
{"columns":[{"label":[""],"name":["_rn_"],"type":[""],"align":["left"]},{"label":["chr"],"name":[1],"type":["chr"],"align":["left"]},{"label":["pos"],"name":[2],"type":["int"],"align":["right"]},{"label":["strand"],"name":[3],"type":["chr"],"align":["left"]},{"label":["Name"],"name":[4],"type":["chr"],"align":["left"]},{"label":["Islands_Name"],"name":[5],"type":["chr"],"align":["left"]},{"label":["Relation_to_Island"],"name":[6],"type":["chr"],"align":["left"]},{"label":["UCSC_RefGene_Name"],"name":[7],"type":["chr"],"align":["left"]},{"label":["UCSC_RefGene_Group"],"name":[8],"type":["chr"],"align":["left"]}],"data":[{"1":"chrY","2":"9363356","3":"-","4":"cg00050873","5":"chrY:9363680-9363943","6":"N_Shore","7":"TSPY4;FAM197Y2","8":"Body;TSS1500","_rn_":"cg00050873"},{"1":"chrY","2":"21239348","3":"-","4":"cg00212031","5":"chrY:21238448-21240005","6":"Island","7":"TTTY14","8":"TSS200","_rn_":"cg00212031"},{"1":"chrY","2":"8148233","3":"-","4":"cg00213748","5":"chrY:8147877-8148210","6":"S_Shore","7":"","8":"","_rn_":"cg00213748"},{"1":"chrY","2":"15815688","3":"-","4":"cg00214611","5":"chrY:15815488-15815779","6":"Island","7":"TMSB4Y;TMSB4Y","8":"1stExon;5'UTR","_rn_":"cg00214611"},{"1":"chrY","2":"9385539","3":"-","4":"cg00455876","5":"chrY:9385471-9385777","6":"Island","7":"","8":"","_rn_":"cg00455876"},{"1":"chrY","2":"6778695","3":"+","4":"cg01707559","5":"chrY:6778574-6780028","6":"Island","7":"TBL1Y;TBL1Y;TBL1Y","8":"TSS200;TSS200;TSS200","_rn_":"cg01707559"}],"options":{"columns":{"min":{},"max":[10]},"rows":{"min":[10],"max":[10]},"pages":{}}}
  </script>
</div>
<p>Merge our significant metaEWAS results with the annotation.</p>
<div class="sourceCode" id="cb222"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">metaEWAS.ann</span><span class="op">&lt;-</span><span class="fu"><a href="https://rdrr.io/pkg/S4Vectors/man/Vector-merge.html">merge</a></span><span class="op">(</span><span class="va">metaEWAS.FDRsig</span>, <span class="va">annotation.table</span>,by.x<span class="op">=</span><span class="st">"probe"</span>,by.y<span class="op">=</span><span class="st">"Name"</span><span class="op">)</span></span>
<span><span class="va">metaEWAS.ann.ord</span><span class="op">&lt;-</span><span class="va">metaEWAS.ann</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/order.html">order</a></span><span class="op">(</span><span class="va">metaEWAS.ann</span><span class="op">$</span><span class="va">p.fe</span><span class="op">)</span>,<span class="op">]</span></span>
<span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/head.html">head</a></span><span class="op">(</span><span class="va">metaEWAS.ann.ord</span><span class="op">)</span></span></code></pre></div>
<div data-pagedtable="false">
<script data-pagedtable-source type="application/json">
{"columns":[{"label":[""],"name":["_rn_"],"type":[""],"align":["left"]},{"label":["probe"],"name":[1],"type":["chr"],"align":["left"]},{"label":["coef_cohort1"],"name":[2],"type":["dbl"],"align":["right"]},{"label":["se_cohort1"],"name":[3],"type":["dbl"],"align":["right"]},{"label":["P_VAL_cohort1"],"name":[4],"type":["dbl"],"align":["right"]},{"label":["N_for_probe_cohort1"],"name":[5],"type":["int"],"align":["right"]},{"label":["coef_cohort2"],"name":[6],"type":["dbl"],"align":["right"]},{"label":["se_cohort2"],"name":[7],"type":["dbl"],"align":["right"]},{"label":["P_VAL_cohort2"],"name":[8],"type":["dbl"],"align":["right"]},{"label":["N_for_probe_cohort2"],"name":[9],"type":["int"],"align":["right"]},{"label":["coef_cohort3"],"name":[10],"type":["dbl"],"align":["right"]},{"label":["se_cohort3"],"name":[11],"type":["dbl"],"align":["right"]},{"label":["P_VAL_cohort3"],"name":[12],"type":["dbl"],"align":["right"]},{"label":["N_for_probe_cohort3"],"name":[13],"type":["int"],"align":["right"]},{"label":["N"],"name":[14],"type":["int"],"align":["right"]},{"label":["coef.fe"],"name":[15],"type":["dbl"],"align":["right"]},{"label":["se.fe"],"name":[16],"type":["dbl"],"align":["right"]},{"label":["p.fe"],"name":[17],"type":["dbl"],"align":["right"]},{"label":["q.fe"],"name":[18],"type":["dbl"],"align":["right"]},{"label":["het.p.fe"],"name":[19],"type":["dbl"],"align":["right"]},{"label":["i2.fe"],"name":[20],"type":["dbl"],"align":["right"]},{"label":["h2.fe"],"name":[21],"type":["dbl"],"align":["right"]},{"label":["fdr"],"name":[22],"type":["dbl"],"align":["right"]},{"label":["chr"],"name":[23],"type":["chr"],"align":["left"]},{"label":["pos"],"name":[24],"type":["int"],"align":["right"]},{"label":["strand"],"name":[25],"type":["chr"],"align":["left"]},{"label":["Islands_Name"],"name":[26],"type":["chr"],"align":["left"]},{"label":["Relation_to_Island"],"name":[27],"type":["chr"],"align":["left"]},{"label":["UCSC_RefGene_Name"],"name":[28],"type":["chr"],"align":["left"]},{"label":["UCSC_RefGene_Group"],"name":[29],"type":["chr"],"align":["left"]}],"data":[{"1":"cg05575921","2":"-0.15723240","3":"0.010576833","4":"3.449766e-31","5":"188","6":"-0.14101120","7":"0.011691620","8":"2.420781e-23","9":"176","10":"-0.15591006","11":"0.010843238","12":"1.583696e-28","13":"166","14":"530","15":"-0.15198539","16":"0.006355165","17":"2.125374e-126","18":"1.2581465","19":"0.5330856","20":"0.00000","21":"0.6290732","22":"8.042838e-122","23":"chr5","24":"373378","25":"+","26":"chr5:373842-374426","27":"N_Shore","28":"AHRR","29":"Body","_rn_":"1934"},{"1":"cg21566642","2":"-0.11908682","3":"0.009286429","4":"8.297569e-26","5":"188","6":"-0.11343605","7":"0.009707079","8":"2.230469e-22","9":"176","10":"-0.10619119","11":"0.009754165","12":"5.891175e-20","13":"166","14":"530","15":"-0.11311143","16":"0.005528416","17":"4.893221e-93","18":"0.9184917","19":"0.6317599","20":"0.00000","21":"0.4592459","22":"9.258464e-89","23":"chr2","24":"233284661","25":"-","26":"chr2:233283397-233285959","27":"Island","28":"","29":"","_rn_":"6541"},{"1":"cg03636183","2":"-0.07262148","3":"0.005700836","4":"1.396044e-25","5":"188","6":"-0.06041403","7":"0.006198676","8":"2.073011e-17","9":"176","10":"-0.07531316","11":"0.006123817","12":"1.881683e-23","13":"166","14":"530","15":"-0.06967483","16":"0.003461447","17":"4.127409e-90","18":"3.3469160","19":"0.1875972","20":"40.24350","21":"1.6734580","22":"5.206314e-86","23":"chr19","24":"17000585","25":"-","26":"chr19:17000627-17001398","27":"N_Shore","28":"F2RL3","29":"Body","_rn_":"1279"},{"1":"cg01940273","2":"-0.07652157","3":"0.006414607","4":"1.992495e-23","5":"188","6":"-0.07737718","7":"0.006436027","8":"3.038132e-23","9":"176","10":"-0.06864217","11":"0.006436418","12":"2.091962e-19","13":"166","14":"530","15":"-0.07418574","16":"0.003711781","17":"7.209404e-89","18":"1.1202947","19":"0.5711249","20":"0.00000","21":"0.5601474","22":"6.820456e-85","23":"chr2","24":"233284934","25":"-","26":"chr2:233283397-233285959","27":"Island","28":"","29":"","_rn_":"715"},{"1":"cg23576855","2":"-0.16467216","3":"0.014655727","4":"1.392884e-21","5":"188","6":"-0.14054494","7":"0.016847122","8":"6.601338e-14","9":"176","10":"-0.18243448","11":"0.013150485","12":"2.632126e-27","13":"166","14":"530","15":"-0.16594019","16":"0.008463186","17":"1.339678e-85","18":"3.8529221","19":"0.1456628","20":"48.09135","21":"1.9264611","22":"1.013922e-81","23":"chr5","24":"373299","25":"-","26":"chr5:373842-374426","27":"N_Shore","28":"AHRR","29":"Body","_rn_":"7030"},{"1":"cg05951221","2":"-0.09316890","3":"0.007577530","4":"2.110764e-24","5":"188","6":"-0.08917528","7":"0.008786733","8":"1.961694e-18","9":"176","10":"-0.08449224","11":"0.007639413","12":"2.190787e-20","13":"166","14":"530","15":"-0.08895022","16":"0.004588173","17":"9.965739e-84","18":"0.6511411","19":"0.7221152","20":"0.00000","21":"0.3255706","22":"6.285392e-80","23":"chr2","24":"233284402","25":"+","26":"chr2:233283397-233285959","27":"Island","28":"","29":"","_rn_":"2070"}],"options":{"columns":{"min":{},"max":[10]},"rows":{"min":[10],"max":[10]},"pages":{}}}
  </script>
</div>
<p>Create list of Genes for Enrichment and save list of FDR CpGs and FDR genes</p>
<div class="sourceCode" id="cb223"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">ann.genes.current</span><span class="op">&lt;-</span><span class="va">metaEWAS.ann.ord</span><span class="op">$</span><span class="va">UCSC_RefGene_Name</span></span>
<span><span class="va">ann.genes.current</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/unlist.html">unlist</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/lapply.html">lapply</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/strsplit.html">strsplit</a></span><span class="op">(</span><span class="va">ann.genes.current</span>, <span class="st">";"</span><span class="op">)</span>, <span class="va">unique</span><span class="op">)</span><span class="op">)</span></span>
<span></span>
<span><span class="co">### 4.4. Save results </span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/write.table.html">write.table</a></span><span class="op">(</span><span class="va">metaEWAS.FDRsig</span><span class="op">$</span><span class="va">probe</span>,<span class="st">"./data/DAY5/Output/list.cpgs.MA.Smoking.NeverVScurrent.txt"</span>, row.names <span class="op">=</span> <span class="cn">F</span>, col.names <span class="op">=</span> <span class="cn">F</span>, quote <span class="op">=</span> <span class="cn">F</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/write.table.html">write.table</a></span><span class="op">(</span><span class="va">metaEWAS.FDRsig</span><span class="op">$</span><span class="va">probe</span><span class="op">[</span><span class="fl">1</span><span class="op">:</span><span class="fl">1000</span><span class="op">]</span>,<span class="st">"./data/DAY5/Output/list.cpgs.MA.Smoking.NeverVScurrent_subset.txt"</span>, row.names <span class="op">=</span> <span class="cn">F</span>, col.names <span class="op">=</span> <span class="cn">F</span>, quote <span class="op">=</span> <span class="cn">F</span><span class="op">)</span></span>
<span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/write.table.html">write.table</a></span><span class="op">(</span><span class="va">ann.genes.current</span>,<span class="st">"./data/DAY5/Output/list.genes.MA.Smoking.NeverVScurrent.txt"</span>, row.names <span class="op">=</span> <span class="cn">F</span>, col.names <span class="op">=</span> <span class="cn">F</span>, quote <span class="op">=</span> <span class="cn">F</span><span class="op">)</span></span></code></pre></div>
</div>
<div id="functional-enrichment-analysis" class="section level2" number="5.3">
<h2>
<span class="header-section-number">5.3</span> Functional enrichment analysis<a class="anchor" aria-label="anchor" href="#functional-enrichment-analysis"><i class="fas fa-link"></i></a>
</h2>
<p>The idea is to compare the list of genes that overlap our CpGs with the list of
all the human genes that are anotated in specific databases. With this, we can see
if our list of genes is a random subset or no.</p>
<p>First of all we convert Gene Symbols to Ensembl and Entrez Gene IDs to use them later</p>
<div class="sourceCode" id="cb224"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">ids</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/clusterProfiler/man/bitr.html">bitr</a></span><span class="op">(</span><span class="va">ann.genes.current</span>, fromType<span class="op">=</span><span class="st">"SYMBOL"</span>, toType<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"ENSEMBL"</span>, <span class="st">"ENTREZID"</span><span class="op">)</span>, OrgDb<span class="op">=</span><span class="st">"org.Hs.eg.db"</span><span class="op">)</span></span></code></pre></div>
<pre><code>## 'select()' returned 1:many mapping between keys and columns</code></pre>
<div id="go" class="section level3" number="5.3.1">
<h3>
<span class="header-section-number">5.3.1</span> GO<a class="anchor" aria-label="anchor" href="#go"><i class="fas fa-link"></i></a>
</h3>
<p>We will first work with the Gene Ontology (GO) database, that allows us to see if a
specific gene function is overrepresented in our gene list. We need to obtain the
list of all human genes that are curated in GO.</p>
<div class="sourceCode" id="cb226"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># GO</span></span>
<span><span class="va">df</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/as.data.frame.html">as.data.frame</a></span><span class="op">(</span><span class="va">org.Hs.egGO</span><span class="op">)</span></span>
<span><span class="va">go_gene_list</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/unique.html">unique</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/sort.html">sort</a></span><span class="op">(</span><span class="va">df</span><span class="op">$</span><span class="va">gene_id</span><span class="op">)</span><span class="op">)</span></span></code></pre></div>
<p>And then run the analyses and keep the most interesting results.</p>
<div class="sourceCode" id="cb227"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">ans.go</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/clusterProfiler/man/enrichGO.html">enrichGO</a></span><span class="op">(</span>gene <span class="op">=</span> <span class="va">ids</span><span class="op">$</span><span class="va">ENTREZID</span>,</span>
<span>                   ont <span class="op">=</span> <span class="st">"BP"</span>,</span>
<span>                   OrgDb <span class="op">=</span><span class="st">"org.Hs.eg.db"</span>,</span>
<span>                   universe <span class="op">=</span> <span class="va">go_gene_list</span>,</span>
<span>                   readable<span class="op">=</span><span class="cn">TRUE</span>,</span>
<span>                   pvalueCutoff <span class="op">=</span> <span class="fl">0.05</span><span class="op">)</span></span>
<span></span>
<span><span class="va">tab.go</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/as.data.frame.html">as.data.frame</a></span><span class="op">(</span><span class="va">ans.go</span><span class="op">)</span></span>
<span><span class="va">tab.go</span><span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/subset.html">subset</a></span><span class="op">(</span><span class="va">tab.go</span>, <span class="va">Count</span><span class="op">&gt;</span><span class="fl">5</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/head.html">head</a></span><span class="op">(</span><span class="va">tab.go</span><span class="op">)</span></span></code></pre></div>
<div data-pagedtable="false">
<script data-pagedtable-source type="application/json">
{"columns":[{"label":[""],"name":["_rn_"],"type":[""],"align":["left"]},{"label":["ID"],"name":[1],"type":["chr"],"align":["left"]},{"label":["Description"],"name":[2],"type":["chr"],"align":["left"]},{"label":["GeneRatio"],"name":[3],"type":["chr"],"align":["left"]},{"label":["BgRatio"],"name":[4],"type":["chr"],"align":["left"]},{"label":["pvalue"],"name":[5],"type":["dbl"],"align":["right"]},{"label":["p.adjust"],"name":[6],"type":["dbl"],"align":["right"]},{"label":["qvalue"],"name":[7],"type":["dbl"],"align":["right"]},{"label":["geneID"],"name":[8],"type":["chr"],"align":["left"]},{"label":["Count"],"name":[9],"type":["int"],"align":["right"]}],"data":[{"1":"GO:0010975","2":"regulation of neuron projection development","3":"152/3833","4":"431/18800","5":"2.935550e-13","6":"1.852039e-09","7":"1.392687e-09","8":"CRTC1/HECW1/RAPGEF1/UBE3A/GFI1/SEMA3F/LYN/GAK/GATA3/CAMK1D/WNT7A/TMEM106B/APOE/EPHA7/BRSK2/COBL/PTK7/NDRG4/RTN4RL1/SYNGAP1/FIG4/FBXO31/MAGI2/BAIAP2/ITM2C/CUX1/SEMA7A/ENC1/CUX2/DNM3/AKT1/PTN/PLXNC1/ARC/CAMK2B/MAG/ARSB/TOX/NFATC4/CD38/IGF1R/EFNB2/STK24/TWF2/NGF/PLXNA4/EPHA4/KNDC1/CDKL3/SEMA5B/SEMA4B/NEDD4L/PLXNA3/DSCAM/ROR1/BCL11A/NTNG2/SEMA3C/CDK5/LRP8/SLC39A12/SEMA6C/STYXL1/RTN4/LTK/DIP2B/WNT3A/MAP2K1/B2M/ULK4/PTK2B/LRP4/KIF13B/ABL2/MIR200C/RHOA/KLK8/SLIT1/TBR1/TRPV2/PTPRD/NLGN1/CFLAR/ULK2/SEMA4C/EHD1/PPP3CA/BRSK1/SEMA3B/PLXND1/BMP7/POU3F2/UST/TRPC5/MARK2/ABI3/DAB1/CFL1/PLXNB1/ATP8A2/NPTN/STK11/NCK1/NTNG1/NEUROG3/KIF1A/CDH2/NRP1/SS18L1/MBP/ALK/PRKD1/PTPRF/SCN1B/HDAC2/DGKG/NFE2L2/EFNB3/CTNNA2/GRN/KLK6/FYN/PDLIM5/PTPRG/KDM1A/ISLR2/EFNA5/CAMK1/TBX6/DAB2/HECW2/PMP22/PACSIN1/PLA2G3/ATP1B2/SEMA4A/FEZ1/EIF4G2/SFRP1/MAP2/SARM1/FZD1/BDNF/SPOCK1/POU4F2/MAP3K13/NTRK1/DIXDC1/TIAM2/NTN1/ROBO2/CAPRIN2","9":"152","_rn_":"GO:0010975"},{"1":"GO:0048568","2":"embryonic organ development","3":"154/3833","4":"449/18800","5":"2.596898e-12","6":"8.191915e-09","7":"6.160116e-09","8":"CDH23/HS6ST1/RARA/PBX1/ZFPM1/DLL1/PAX6/HIPK2/KCNQ1/POLE/CHD7/HLX/DNAJB6/LBX1/COL11A1/TRIOBP/TBX1/GATA3/MYO3B/TGFB3/COBL/PTK7/NDRG4/CCDC40/TFEB/PRKRA/SPINT2/DSCAML1/KCNQ4/VPS52/TNF/RARRES2/MMP14/HOXC11/HAND2/PDGFRA/AKT1/PRDM1/SH2B3/HPN/KDM2B/SALL1/ASCL2/TIFAB/OSR1/GSC/SATB2/NES/HOXB3/USH1C/PLXNA4/HOXA11/LRIG3/TPO/HOXC4/LEF1/RUNX2/BMPR1A/WNT2/HOXA4/PAX8/DLX6/HIPK1/CSF2/IFT122/POU3F4/GLI1/SLC44A4/COL13A1/WNT3A/TTC39C/PDZD7/LHX1/SP3/FZD5/MAP2K1/HOXA7/ASB2/VASH2/HOXA2/RARB/CDX4/IFT140/HOXD9/TGFBR2/TFAP2A/HOXA9/PBX2/GATA4/PAX2/TWIST1/CHST11/TBX20/RARG/CDKN1C/SOX18/NEUROG1/ECE1/TSHZ1/SOX15/TEAD2/PITX2/FZD6/BMP7/PKD2/CTHRC1/HEY1/CDC42/MYF5/SMO/PPP1R13L/ATP8A2/CASP8/CITED1/CRB2/AHI1/HOXB5/GATA2/RPL10/FBN2/SOBP/HOXB2/ID3/TP53/ALDH1A3/MYO3A/FOXG1/GRHL3/COL2A1/BLOC1S2/TUBB2B/ACVR1/GRHL2/HNF1B/TGFB2/FGF8/TBX6/EPAS1/PHACTR4/IFT172/TMED2/C2CD3/HOXB8/ATF4/RBPJ/GLI2/PDGFA/NR2F2/NSDHL/IGF2/NTN1/WNT7B/PKD1/GLI3","9":"154","_rn_":"GO:0048568"},{"1":"GO:0007409","2":"axonogenesis","3":"148/3833","4":"430/18800","5":"5.142088e-12","6":"8.310101e-09","7":"6.248989e-09","8":"PTK2/PALLD/PAX6/EPHA6/SEMA3F/GBX1/GATA3/WNT7A/APOE/EPHA7/BRSK2/COBL/NKX6-1/SYNGAP1/NFIB/BAIAP2/ARTN/DSCAML1/KLF7/SEMA7A/PARD3/NRXN1/PLXNC1/MAG/CTTN/IGF1R/APLP2/ITGB1/EFNB2/BCL2/TRIO/TWF2/NGF/PLXNA4/EPHA4/CDKL3/SEMA5B/SEMA4B/ROBO3/PLXNA3/DSCAM/NTNG2/SEMA3C/MAP1S/CDK5/ACTB/SEMA6C/APP/RTN4/SLITRK3/DIP2B/EXT1/WNT3A/LHX1/CDK5R2/LMX1A/FLOT1/ISL1/MAP2K1/CDH11/LRP4/KIF13B/MAP1A/ALCAM/HOXA2/NEO1/SPG11/SLIT1/TBR1/TRPV2/EGR2/ARX/ULK2/LAMB2/PAX2/SEMA4C/CNTNAP1/ECE1/BRSK1/SEMA3B/PLXND1/NUMBL/BMP7/LGR6/POU3F2/UST/TRPC5/EFNB1/MARK2/DAB1/RAC1/GFRA3/BCL11B/PLXNB1/VCL/SMO/ATP8A2/BOC/FLRT2/NPTN/STK11/TOP2B/EFNA3/NTNG1/CDH2/SLC9A6/NRP1/MBP/CNP/PTPRM/AUTS2/SCN1B/EPHA8/EFNB3/CTNNA2/SLIT3/FOXG1/FYN/YTHDF1/NRXN3/SPTBN4/TUBB2B/ISLR2/EFNA5/PRKCQ/BARHL2/ERBB2/TAOK2/KIF5C/SEMA4A/FEZ1/NOTCH3/EIF4G2/GLI2/MAP2/DPYSL2/KALRN/BDNF/POU4F2/MAP3K13/NTRK1/DIXDC1/NRP2/TIAM2/NTN1/ROBO2/LGI1/GLI3","9":"148","_rn_":"GO:0007409"},{"1":"GO:0061564","2":"axon development","3":"161/3833","4":"479/18800","5":"5.268728e-12","6":"8.310101e-09","7":"6.248989e-09","8":"PTK2/PALLD/PAX6/EPHA6/SEMA3F/GBX1/GATA3/WNT7A/APOE/EPHA7/BRSK2/COBL/NKX6-1/RTN4RL1/SYNGAP1/NFIB/BAIAP2/ARTN/DSCAML1/KLF7/SEMA7A/PARD3/NRXN1/PTN/PLXNC1/MAG/CRTAC1/CTTN/IGF1R/APLP2/ITGB1/EFNB2/BCL2/STK24/TRIO/TWF2/NGF/PLXNA4/EPHA4/CDKL3/SEMA5B/DDR1/SEMA4B/CSPG5/ROBO3/NCAM2/PLXNA3/DSCAM/NTNG2/SEMA3C/MAP1S/CDK5/ACTB/SEMA6C/APP/RTN4/SLITRK3/DIP2B/EXT1/WNT3A/LHX1/MTR/CDK5R2/LMX1A/FLOT1/ISL1/MAP2K1/CDH11/LRP4/KIF13B/MAP1A/ALCAM/VASH2/HOXA2/NEO1/SPG11/KLK8/SLIT1/TBR1/TRPV2/EGR2/ARX/ULK2/LAMB2/PAX2/SEMA4C/CNTNAP1/ECE1/BRSK1/SEMA3B/PLXND1/NUMBL/BMP7/MAPK8IP3/LGR6/POU3F2/UST/TRPC5/EFNB1/MARK2/DAB1/RAC1/GFRA3/BCL11B/PLXNB1/VCL/SMO/ATP8A2/BOC/FLRT2/NPTN/STK11/TOP2B/EFNA3/NTNG1/CDH2/SLC9A6/NRP1/MBP/CNP/PTPRM/PTPRF/AUTS2/SCN1B/EPHA8/EFNB3/CTNNA2/SLIT3/GRN/FOXG1/FYN/YTHDF1/NRXN3/SPTBN4/TUBB2B/ISLR2/EFNA5/PRKCQ/BARHL2/ERBB2/TAOK2/KIF5C/SEMA4A/FEZ1/NOTCH3/EIF4G2/GLI2/MAP2/DPYSL2/KALRN/BDNF/POU4F2/MAP3K13/NTRK1/DIXDC1/NRP2/TIAM2/NTN1/ROBO2/LGI1/GLI3","9":"161","_rn_":"GO:0061564"},{"1":"GO:0060284","2":"regulation of cell development","3":"163/3833","4":"500/18800","5":"5.900166e-11","6":"7.444829e-08","7":"5.598326e-08","8":"DLL1/PAX6/SEMA3F/LYN/CHD7/HAP1/DLL3/TTBK1/TRIOBP/WNT7A/EPHA7/NKX6-1/SYNGAP1/PLCB1/DAAM2/FBXO31/PDE5A/BAIAP2/IL6/TNF/CUX1/SEMA7A/CUX2/PRKDC/TNFRSF1B/AKT1/PTN/PLXNC1/CAMK2B/ILK/MAG/SPRY4/PRPF19/GPR68/NFATC4/ASCL2/CXCR4/BCL2/PRKCH/RGS14/ARHGEF7/CX3CR1/ARPC2/HOXB3/TWF2/NGF/PLXNA4/HOXA11/EPHA4/IGF1/CDKL3/SEMA5B/SEMA4B/FBXO22/NKX2-2/CSPG5/SKI/BMPR1A/PLXNA3/DSCAM/BCL11A/WNT2/SEMA3C/P4HB/HDAC9/DYNLT1/CDK5/LRP8/GSX2/LIMS2/SOX10/ID4/SLAMF8/SEMA6C/RTN4/DIP2B/PCM1/WNT3A/MAP2K1/B2M/LRP4/DLL4/SLIT1/AURKA/TRPV2/PTPRD/INS/EGR2/SORL1/ULK2/ACTN4/SEMA4C/PPP3CA/EDNRB/MPL/SEMA3B/PLXND1/NUMBL/BMP7/TYROBP/HEY1/MYADM/TRPC5/LTA/PRMT5/CDC42/DAB1/MYF5/TP73/DOCK5/RAC1/PLXNB1/VCL/SMO/NPTN/STK11/AXIN2/NRP1/SS18L1/MBP/FBLN1/IKBKB/HDAC2/SMARCD3/TP53/PITX3/PTHLH/EFNB3/GRN/FOXG1/BIN1/SERPINE2/KDM1A/ROCK2/LIMS1/HAS2/ISLR2/EFNA5/LDLR/TBX6/DAB2/CLCF1/SEMA4A/EIF4G2/PRTG/MAP2/FAIM/BDNF/POU4F2/PDE3A/MAP3K13/RASSF10/TNFRSF1A/DIXDC1/TIAM2/NTN1/MIR181C/ROBO2/TRIB1/SRRT/GLI3/F2/CAPRIN2","9":"163","_rn_":"GO:0060284"},{"1":"GO:0001503","2":"ossification","3":"141/3833","4":"420/18800","5":"1.219377e-10","6":"1.282175e-07","7":"9.641638e-08","8":"LRP5/PBX1/XYLT1/PTK2/CDK6/MINPP1/SMAD6/TP63/COL11A1/RUNX3/TGFB3/TMEM119/CTNNBIP1/SBNO2/IL6/CHRDL1/TNF/SEMA7A/MMP14/FGF23/IL6R/HAND2/AKT1/SMOC1/PTN/IGFBP5/ILK/CRIM1/BCL2/OSR1/CLEC3B/RRAS2/SLC24A3/SATB2/IFT80/ZBTB16/IGF1/VCAN/ADRB2/LEF1/SKI/RUNX2/BMPR1A/ID4/ISG15/MEF2D/RDH14/GLI1/PPARGC1B/HDAC7/MIR208A/EIF2AK3/TMEM38B/IFITM1/EXT1/COL13A1/WNT3A/SORT1/SP3/HIRA/SYNCRIP/PDLIM7/CDH11/PTK2B/LRP4/MGP/HOXA2/MIR200C/COL1A1/RHOA/ACHE/FAM20C/TFAP2A/HSD17B4/RRBP1/EGR2/KLF10/TWIST1/FBXO5/S1PR1/FGF18/FASN/BMP7/KAZALD1/P2RX7/CTHRC1/KL/MYF5/PHOSPHO1/WWTR1/PLXNB1/SMO/CHRD/COL6A1/CITED1/GIT1/AXIN2/DMP1/RORB/PRKD1/FBN2/IBSP/ID3/TRPM4/LEP/MATN1/PTHLH/MMP9/ANO6/SOX2/COL2A1/CBFB/PENK/FGF2/FBXL15/ACVR1/LIMD1/TGFB2/HNRNPC/BMP8B/SMPD3/KREMEN2/MIR20A/ATF4/NAB2/GABBR1/RBPJ/RSPO2/GLI2/MIR138-1/SFRP1/BGLAP/ALOX5/FZD1/IGF2/SNRNP200/BMP6/WNT7B/RASSF2/THRA/GLI3","9":"141","_rn_":"GO:0001503"}],"options":{"columns":{"min":{},"max":[10]},"rows":{"min":[10],"max":[10]},"pages":{}}}
  </script>
</div>
<p>Finally we can perform different type of plots to see the results in a graphical way</p>
<div class="sourceCode" id="cb228"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">p1</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/graphics/barplot.html">barplot</a></span><span class="op">(</span><span class="va">ans.go</span>, showCategory<span class="op">=</span><span class="fl">10</span><span class="op">)</span> <span class="op">+</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/labs.html">ggtitle</a></span><span class="op">(</span><span class="st">'Never vs Current Smokers'</span><span class="op">)</span> <span class="op">+</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/theme.html">theme</a></span><span class="op">(</span>plot.title <span class="op">=</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/element.html">element_text</a></span><span class="op">(</span>size <span class="op">=</span> <span class="fl">18</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="va">p1</span></span></code></pre></div>
<div class="inline-figure"><img src="plots/goplotss-1.png" width="576" style="display: block; margin: auto;"></div>
<div class="sourceCode" id="cb229"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">ans.go</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/enrichplot/man/pairwise_termsim.html">pairwise_termsim</a></span><span class="op">(</span><span class="va">ans.go</span><span class="op">)</span></span>
<span><span class="va">p2</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/enrichplot/man/emapplot.html">emapplot</a></span><span class="op">(</span><span class="va">ans.go</span>, cex_label_category <span class="op">=</span> <span class="fl">0.5</span>, showCategory <span class="op">=</span> <span class="fl">20</span><span class="op">)</span> <span class="op">+</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/labs.html">ggtitle</a></span><span class="op">(</span><span class="st">'Never vs Current Smokers'</span><span class="op">)</span> <span class="op">+</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/theme.html">theme</a></span><span class="op">(</span>plot.title <span class="op">=</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/element.html">element_text</a></span><span class="op">(</span>size <span class="op">=</span> <span class="fl">18</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="va">p2</span></span></code></pre></div>
<div class="inline-figure"><img src="plots/goplotss-2.png" width="576" style="display: block; margin: auto;"></div>
</div>
<div id="reactome" class="section level3" number="5.3.2">
<h3>
<span class="header-section-number">5.3.2</span> Reactome<a class="anchor" aria-label="anchor" href="#reactome"><i class="fas fa-link"></i></a>
</h3>
<p>Now we do the same for the Reactome database</p>
<div class="sourceCode" id="cb230"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">ans.react</span> <span class="op">&lt;-</span> <span class="fu">enrichPathway</span><span class="op">(</span>gene<span class="op">=</span><span class="va">ids</span><span class="op">$</span><span class="va">ENTREZID</span>, pvalueCutoff <span class="op">=</span> <span class="fl">0.05</span>, readable<span class="op">=</span><span class="cn">TRUE</span><span class="op">)</span></span>
<span><span class="va">tab.react</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/as.data.frame.html">as.data.frame</a></span><span class="op">(</span><span class="va">ans.react</span><span class="op">)</span></span>
<span><span class="va">tab.react</span><span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/subset.html">subset</a></span><span class="op">(</span><span class="va">tab.react</span>, <span class="va">Count</span><span class="op">&gt;</span><span class="fl">5</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/head.html">head</a></span><span class="op">(</span><span class="va">tab.react</span><span class="op">)</span></span></code></pre></div>
</div>
<div id="disease" class="section level3" number="5.3.3">
<h3>
<span class="header-section-number">5.3.3</span> Disease<a class="anchor" aria-label="anchor" href="#disease"><i class="fas fa-link"></i></a>
</h3>
<p>To be provided</p>
</div>
<div id="online-tools" class="section level3" number="5.3.4">
<h3>
<span class="header-section-number">5.3.4</span> Online Tools<a class="anchor" aria-label="anchor" href="#online-tools"><i class="fas fa-link"></i></a>
</h3>
<p>The functional enrichment can also be done online with some interesting tools:</p>
<p><a href="https://eforge.altiusinstitute.org/"><strong>eFORGE</strong></a> is used to identify cell-type
or tissue-type specific signals in epigenomic data by looking at the overlap between
differentially methylated positions (DMPs) with DNase I hypersensitive sites (DHSs)</p>
<p><a href="http://www.ewascatalog.org/"><strong>EWAS Catalog</strong></a> is a huge database of EWAS results.
We can submit the names of our top CpGs to see the last published information about them</p>
<p><strong>ANSWERS</strong></p>
<ol style="list-style-type: decimal">
<li>For current smokers it is the regulation of neuron projection development. For former smokers it is</li>
</ol>
</div>
</div>
</div>
  <div class="chapter-nav">
<div class="prev"><a href="meta-analysis-of-epigenome-wide-association-studies-metaewas.html"><span class="header-section-number">4</span> Meta-analysis of epigenome-wide association studies (metaEWAS)</a></div>
<div class="next"><a href="references.html">References</a></div>
</div></main><div class="col-md-3 col-lg-2 d-none d-md-block sidebar sidebar-chapter">
    <nav id="toc" data-toggle="toc" aria-label="On this page"><h2>On this page</h2>
      <ul class="nav navbar-nav">
<li><a class="nav-link" href="#biological-interpretation-of-findings"><span class="header-section-number">5</span> Biological interpretation of findings</a></li>
<li><a class="nav-link" href="#load-metaewas-results"><span class="header-section-number">5.1</span> Load MetaEWAS results</a></li>
<li><a class="nav-link" href="#annotation-of-cpg-sites"><span class="header-section-number">5.2</span> Annotation of CpG sites</a></li>
<li>
<a class="nav-link" href="#functional-enrichment-analysis"><span class="header-section-number">5.3</span> Functional enrichment analysis</a><ul class="nav navbar-nav">
<li><a class="nav-link" href="#go"><span class="header-section-number">5.3.1</span> GO</a></li>
<li><a class="nav-link" href="#reactome"><span class="header-section-number">5.3.2</span> Reactome</a></li>
<li><a class="nav-link" href="#disease"><span class="header-section-number">5.3.3</span> Disease</a></li>
<li><a class="nav-link" href="#online-tools"><span class="header-section-number">5.3.4</span> Online Tools</a></li>
</ul>
</li>
</ul>

      <div class="book-extra">
        
        <ul class="list-unstyled">
          
        </ul>
</div>
    </nav>
</div>
  

</div>
</div> <!-- .container -->

<footer class="bg-primary text-light mt-5"><div class="container"><div class="row">

  <div class="col-12 col-md-6 mt-3">
    <p>"<strong>Introduction to Epigenome-Wide Association Studies (EWAS)</strong>" was written by Juan R. González, Mariona Bustamante, Sofía Aguilar &amp; Laura Balagué. It was last built on 2023-09-02.</p>
  </div>

  <div class="col-12 col-md-6 mt-3">
    <p>This book was built by the <a class="text-light" href="https://bookdown.org">bookdown</a> R package.</p>
  </div>

</div></div>
</footer><!-- dynamically load mathjax for compatibility with self-contained --><script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    var src = "true";
    if (src === "" || src === "true") src = "https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.9/latest.js?config=TeX-MML-AM_CHTML";
    if (location.protocol !== "file:")
      if (/^https?:/.test(src))
        src = src.replace(/^https?:/, '');
    script.src = src;
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script><script type="text/x-mathjax-config">const popovers = document.querySelectorAll('a.footnote-ref[data-toggle="popover"]');
for (let popover of popovers) {
  const div = document.createElement('div');
  div.setAttribute('style', 'position: absolute; top: 0, left:0; width:0, height:0, overflow: hidden; visibility: hidden;');
  div.innerHTML = popover.getAttribute('data-content');

  var has_math = div.querySelector("span.math");
  if (has_math) {
    document.body.appendChild(div);
    MathJax.Hub.Queue(["Typeset", MathJax.Hub, div]);
    MathJax.Hub.Queue(function() {
      popover.setAttribute('data-content', div.innerHTML);
      document.body.removeChild(div);
    })
  }
}
</script>
</body>
</html>
